frame/balances/src/lib.rs

```diff
@@ -1189,12 +1189,15 @@ impl<T: Config<I>, I: 'static> fungible::MutateHold<T::AccountId> for Pallet<T,
 		if amount.is_zero() {
 			return Ok(())
 		}
-		ensure!(Self::can_reserve(who, amount), Error::<T, I>::InsufficientBalance);
-		Self::mutate_account(who, |a| {
-			a.free -= amount;
-			a.reserved += amount;
-		})?;
-		Ok(())
+		ensure!(
+			<Self as fungible::InspectHold<T::AccountId>>::can_hold(who, amount),
```

Comment:
Okay, after writing unit tests, I found out that there is a difference between `can_reserve` and `can_hold`: the latter ensures that the reserved/held amount doesn't make the free balance go below the existential deposit. Should we still continue using `can_hold` here then?

Reply:
if we change `can_hold` what that mess up any existing code using `can_hold` / what are the tradeoffs?

Reply:
We want people to be able to use fungible traits instead of `Currency` in the future. See #8285 and #8453. We currently use `Currency` traits everywhere, and we'd like to change that.